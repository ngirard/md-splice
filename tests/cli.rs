use assert_cmd::Command;
use insta::assert_snapshot;

fn cmd() -> Command {
    Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap()
}

#[test]
fn test_i1_version_flag() {
    let output = cmd().arg("--version").output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_version", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag() {
    let output = cmd().arg("--help").output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag_insert() {
    let output = cmd().args(["insert", "--help"]).output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help_insert", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag_replace() {
    let output = cmd().args(["replace", "--help"]).output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help_replace", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}
