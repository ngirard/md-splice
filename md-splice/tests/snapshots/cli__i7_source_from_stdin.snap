---
source: tests/cli.rs
expression: stdout
---
# Source from STDIN

This paragraph was replaced via STDIN.
